finality-aleph/src/party/mod.rs

```diff
@@ -470,7 +486,52 @@ where
         };
         loop {
             let last_finalized_number = self.client.info().finalized_number;
-            debug!(target: "afa", "Highest finalized: {:?} session {:?}", last_finalized_number, session_id);
+            if last_finalized_number > first_block {
```

Comment:
Yes. @timorl am I right it is ok?

Reply:
The advantage is that this might catch e.g. a public address change early and improve the chances of being connected when the time comes. So I would say yes, especially since this shouldn't be a costly operation.

Reply:
But the difference between calling it once, vs a few times is only that `self.session_manager.start_validator_session` will be called once or a few times. Right?

But I would definitely expect this `start_validator_session` to be idemponent.

Reply:
It reads the currently known external addresses and peer id, these can theoretically change (at least the former).

Reply:
Also, one tiny detail: imho this should be `last_finalized_number >= first_block`. The next_authorities should be there in the first block of the new_session already.